File: backend/FragranceCommerce.Api/Models/Order.cs
namespace FragranceCommerce.Api.Models;

using FragranceCommerce.Api.Enums;

public class Order : BaseEntity
{
    public Guid UserId { get; set; }
    public User User { get; set; } = null!;
    public string OrderNumber { get; set; } = string.Empty;
    public decimal TotalAmount { get; set; }
    public string? CouponCode { get; set; }
    public decimal DiscountAmount { get; set; }
    public decimal FinalAmount { get; set; }
    public OrderStatus Status { get; set; } = OrderStatus.Pending;
    public DateTime OrderedAt { get; set; } = DateTime.UtcNow;
    public OrderAddress? ShippingAddress { get; set; }
    public Payment? Payment { get; set; }
    public ICollection<OrderItem> Items { get; set; } = new List<OrderItem>();
}

File: frontend/fragrance-commerce-web/src/types/order.ts
export interface CreateOrderRequest {
    addressId: string;
    paymentMethod: string;
}

export interface OrderAddress {
    fullName: string;
    phoneNumber: string;
    addressLine1: string;
    addressLine2: string;
    city: string;
    state: string;
    postalCode: string;
    country: string;
}

export interface OrderPayment {
    id: string;
    amount: number;
    paymentMethod: string;
    paymentStatus: string;
    transactionId: string | null;
    paidAt: string | null;
}

export interface OrderItem {
    id: string;
    productVariantId: string;
    productName: string;
    variantName: string;
    unitPrice: number;
    quantity: number;
    totalPrice: number;
}

export interface Order {
    id: string;
    orderNumber: string;
    status: string;
    orderedAt: string;
    totalAmount: number;
    couponCode: string | null;
    discountAmount: number;
    finalAmount: number;
    shippingAddress: OrderAddress;
    payment: OrderPayment;
    items: OrderItem[];
}